cdap-watchdog/src/main/java/co/cask/cdap/logging/write/AvroFileWriter.java

```diff
@@ -174,9 +182,25 @@ private AvroFile createAvroFile(LoggingContext loggingContext, long timestamp) t
     return avroFile;
   }
 
-  private Location createLocation(String pathFragment, long timestamp) throws IOException {
+  private Location createLocation(LoggingContext loggingContext, long timestamp) throws IOException {
+    Map<String, LoggingContext.SystemTag> systemTags = loggingContext.getSystemTagsMap();
+    LoggingContext.SystemTag topLevelTag;
+    if (systemTags.containsKey(NamespaceLoggingContext.TAG_NAMESPACE_ID)) {
+      topLevelTag = systemTags.get(NamespaceLoggingContext.TAG_NAMESPACE_ID);
+    } else if (systemTags.containsKey(SystemLoggingContext.TAG_SYSTEM_ID)) {
+      topLevelTag = systemTags.get(SystemLoggingContext.TAG_SYSTEM_ID);
+    } else {
+      throw new IllegalArgumentException(
+        String.format("Expected top level tag in logging context to be either %s or %s. Path fragment = %s.",
+                      NamespaceLoggingContext.TAG_NAMESPACE_ID, SystemLoggingContext.TAG_SYSTEM_ID,
+                      loggingContext.getLogPathFragment()));
+    }
+    String namespace = topLevelTag.getValue();
+    String logBaseDir = cConf.get(LoggingConfiguration.LOG_BASE_DIR);
+    String pathFragment = loggingContext.getLogPathFragment().replaceAll(String.format("%s/", namespace), "");
```

Comment:
Should we instead change `getLogPathFragment()` to take in a base dir and then give the right path? Or, do away with `getLogPathFragment()`, as it does not seem to be adding any value.